pkg/otlp/config.go

```diff
@@ -67,30 +68,34 @@ func readConfigSection(cfg config.Config, section string) *confmap.Conf {
 }
 
 // FromAgentConfig builds a pipeline configuration from an Agent configuration.
-func FromAgentConfig(cfg config.Config) (PipelineConfig, error) {
+func FromAgentConfig(cfg config.Component) (PipelineConfig, error) {
 	var errs []error
-	otlpConfig := readConfigSection(cfg, config.OTLPReceiverSection)
-	censusConfig := readConfigSection(cfg, config.OTLPCensusReceiverSection)
+	otlpConfig := readConfigSection(cfg, coreconfig.OTLPReceiverSection)
+	censusConfig := readConfigSection(cfg, coreconfig.OTLPCensusReceiverSection)
 
-	tracePort, err := portToUint(cfg.GetInt(config.OTLPTracePort))
+	tracePort, err := portToUint(cfg.GetInt(coreconfig.OTLPTracePort))
 	if err != nil {
 		errs = append(errs, fmt.Errorf("internal trace port is invalid: %w", err))
 	}
-
-	metricsEnabled := cfg.GetBool(config.OTLPMetricsEnabled)
-	tracesEnabled := cfg.GetBool(config.OTLPTracesEnabled)
-	logsEnabled := cfg.GetBool(config.OTLPLogsEnabled)
+	metricsEnabled := cfg.GetBool(coreconfig.OTLPMetricsEnabled)
+	tracesEnabled := cfg.GetBool(coreconfig.OTLPTracesEnabled)
+	logsEnabled := cfg.GetBool(coreconfig.OTLPLogsEnabled)
 	if !metricsEnabled && !tracesEnabled {
 		errs = append(errs, fmt.Errorf("at least one OTLP signal needs to be enabled"))
 	}
-	metricsConfig := readConfigSection(cfg, config.OTLPMetrics)
-	debugConfig := readConfigSection(cfg, config.OTLPDebug)
+	metricsConfig := readConfigSection(cfg, coreconfig.OTLPMetrics)
+	debugConfig := readConfigSection(cfg, coreconfig.OTLPDebug)
 
+	health := coreconfig.DefaultOTELHealthEndpoint
+	if v := cfg.GetString(coreconfig.OTLPHealthEndpoint); v != "" {
+		health = v
+	}
```

Comment:
Instead of doing this, set a default value on the configuration component. This is what we do for all the other configuration toggles.

Reply:
I think I already do that [here](https://github.com/DataDog/datadog-agent/blob/7e356018c54bac6855f8d6a5eb6a08cf6a3b8206/pkg/config/otlp.go#L109). This was needed for some test case. I can double check. I can remove it if you like but wanted to play it safe. LMK what you think.

Reply:
We don't do this for any other configuration setting, so I would be suprised if we need it for this one specifically

Reply:
Done.